Employee overview2

ParkwayBE opened the pull request.

ParkwayBE commented:
i need the user_id in the employee model

ThomasFrans commented:
You need to merge main into your branch before it can be merged as it has merge conflicts (and please **not with a PR on GitHub**, just merge it locally).

ParkwayBE commented:
> You need to merge main into your branch before it can be merged as it has merge conflicts (and please **not with a PR on GitHub**, just merge it locally)
okay i did that.

ThomasFrans commented:
https://github.com/Practice-Enterprise2/AD#pull-request-requirements

ParkwayBE commented:
> https://github.com/Practice-Enterprise2/AD#pull-request-requirements

i removed the commented out stuff and i removed the css. I also formatted the code.

ThomasFrans commented:
You don't have to remove the CSS, only in components/layouts where it can interfere with other parts of the UI.

ThomasFrans commented on app/Models/Employee.php:
```diff
@@ -44,6 +44,7 @@ class Employee extends Model implements ValidatesAttributes
     protected $table = 'employees';
 
     protected $fillable = [
+        'user_id',
```
Not mass fillable

ThomasFrans commented on app/Models/User.php:
```diff
@@ -72,6 +72,7 @@ class User extends Authenticatable implements MustVerifyEmail, ValidatesAttribut
     ];
 
     protected $fillable = [
+        'address_id',
```
Not mass fillable

ThomasFrans commented on resources/views/add_employee.blade.php:
```diff
@@ -1,5 +1,48 @@
 <x-app-layout>
-  <x-employees.create />
```
This was turned into a Blade component so it could also be included on other pages (like in the control panel). If it is a component it becomes much more reusable.

ThomasFrans commented on resources/views/layouts/navigation.blade.php:
```diff
@@ -23,9 +23,28 @@ class="block h-9 w-auto fill-current text-gray-800 dark:text-gray-200" />
-            <x-nav-link :href="route('employee')" :active="request()->routeIs('employee')">
-              {{ __('Employee') }}
-            </x-nav-link>
+            <x-dropdown>
+              <x-slot name="trigger">
+                <button
+                  class="mt-4 inline-flex items-center rounded-md border border-transparent border-white bg-white px-3 py-2 text-sm font-medium leading-4 text-gray-500 transition duration-150 ease-in-out hover:text-gray-700 focus:outline-none dark:bg-gray-800 dark:text-gray-400 dark:hover:text-gray-300">
+                  Employee
+                  <div class="ml-1">
+                    <svg class="h-4 w-4 fill-current" viewBox="0 0 20 20">
+                      <path fill-rule="evenodd"
+                        d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
+                        clip-rule="evenodd" />
+                    </svg>
+
+                  </div>
+                </button>
+              </x-slot>
+
+              <x-slot name="content">
+                <x-dropdown-link :href="route('employee.overview')" :active="request()->routeIs('employee')">>
+                  {{ __('Employee Overview') }}
+                </x-dropdown-link>
+              </x-slot>
+            </x-dropdown>
```
The navigation is already too full with stuff. Place this somewhere where it makes sense (as Jente discussed last Thursday). Look on the UI/UX flows in the Discord.

ParkwayBE commented:
This should be good to go now.

ParkwayBE commented:
I changed what you have told me to change. so this should be good now.

ThomasFrans commented on resources/views/components/employees/create.blade.php:
```diff
@@ -1,61 +1,93 @@
-    <button type="submit"
-      class="bg-red m-[50px] w-[200px] rounded-[250px] p-[15px] text-2xl hover:bg-violet-500 hover:text-black">New</button>
-  </form>
+<style>
+  .employee-add-fix:after {
+    visibility: hidden;
+    display: block;
+    font-size: 0;
+    content: " ";
+    clear: both;
+    height: 0;
+  }
+</style>
```
https://github.com/Practice-Enterprise2/AD#pull-request-requirements

ThomasFrans commented on routes/web.php:
```diff
@@ -60,10 +58,13 @@
-        Route::post('/employee_add', 'save');
+        Route::get('/employee_overview', 'index')->name('employee.overview');
+        Route::post('/employee_add', 'save')->name('save-employee');
+        Route::get('/new_employee', 'showAdd')->name('employee.create');
```
This route used policies to authorize the action. Why is it removed?

ParkwayBE replied:
sorry I dont understand what should be changed here. can you tell me what should be there instead?

ThomasFrans replied:
In the diff it shows that there used to be middleware for authorization.
